jax/_src/interpreters/pxla.py

```diff
@@ -1683,7 +1683,10 @@ class DeviceAssignmentMismatchError(Exception):
 
 
 def _get_default_device() -> xc.Device:
-  return config.default_device.value or xb.local_devices()[0]
+  if isinstance(config.default_device.value, str):
+    return xb.get_backend(config.default_device.value).devices()[0]
```

Comment:
This should be `.local_devices()[0]` otherwise it won't work in multi-controller JAX.

You should add a test for this internally. I can point you to the file (but not in this comment since the file is internal)